libfirm-rs/src/nodes_gen.rs

```diff
@@ -0,0 +1,4048 @@
+                    _ => ProjKind::Other,
+               },
+            Node::Builtin(node) => 
+               match proj.num() {
+                    0 => ProjKind::Builtin_M(node),
+                    _ => ProjKind::Other,
+               },
+            Node::Call(node) => 
+               match proj.num() {
+                    0 => ProjKind::Call_M(node),
+                    1 => ProjKind::Call_TResult(node),
+                    2 => ProjKind::Call_XRegular(node),
+                    3 => ProjKind::Call_XExcept(node),
+                    _ => ProjKind::Other,
+               },
+            Node::Cond(node) => 
+               match proj.num() {
+                    0 => ProjKind::Cond_False(node),
+                    1 => ProjKind::Cond_True(node),
+                    _ => ProjKind::Other,
+               },
+            Node::Div(node) => 
+               match proj.num() {
+                    0 => ProjKind::Div_M(node),
+                    1 => ProjKind::Div_Res(node),
+                    2 => ProjKind::Div_XRegular(node),
+                    3 => ProjKind::Div_XExcept(node),
+                    _ => ProjKind::Other,
+               },
+            Node::Load(node) => 
+               match proj.num() {
+                    0 => ProjKind::Load_M(node),
+                    1 => ProjKind::Load_Res(node),
+                    2 => ProjKind::Load_XRegular(node),
+                    3 => ProjKind::Load_XExcept(node),
+                    _ => ProjKind::Other,
+               },
+            Node::Mod(node) => 
+               match proj.num() {
+                    0 => ProjKind::Mod_M(node),
+                    1 => ProjKind::Mod_Res(node),
+                    2 => ProjKind::Mod_XRegular(node),
+                    3 => ProjKind::Mod_XExcept(node),
+                    _ => ProjKind::Other,
+               },
+            Node::Raise(node) => 
+               match proj.num() {
+                    0 => ProjKind::Raise_M(node),
+                    1 => ProjKind::Raise_X(node),
+                    _ => ProjKind::Other,
+               },
+            Node::Start(node) => 
+               match proj.num() {
+                    0 => ProjKind::Start_M(node),
+                    1 => ProjKind::Start_PFrameBase(node),
+                    2 => ProjKind::Start_TArgs(node),
+                    _ => ProjKind::Other,
+               },
+            Node::Store(node) => 
+               match proj.num() {
+                    0 => ProjKind::Store_M(node),
+                    1 => ProjKind::Store_XRegular(node),
+                    2 => ProjKind::Store_XExcept(node),
+                    _ => ProjKind::Other,
+               },
+            Node::Switch(node) => 
+               match proj.num() {
+                    0 => ProjKind::Switch_Default(node),
+                    _ => ProjKind::Other,
+               },
+            Node::Proj(proj, ProjKind::Start_TArgs(start)) => ProjKind::Start_TArgs_Arg(proj.num(), start, proj),
```

Comment:
possible endless recursion

Reply:
per discussion with andreas and sebastian today, this is guranteed to be impossible by libfirm internals.